harvdev_utils/general_functions/checks.py

```diff
@@ -0,0 +1,136 @@
+"""Check for proforma log and xml files."""
+from collections import deque
+import re
+
+
+def xort_logfile_check(filename):
+    """Check output from xort.
+
+    filename: full filename and path of xort log file.
+              will normally be something like
+              /users/falls/DB_update/epicycle_log/harv_2017_02_e5_p1.load
+
+    return wether okay or not and an error string
+    """
+    okay = True
+    errors = []
+
+    # check log file for the word ERROR
+    count = 0
+    old_lines = deque(maxlen=10)
+    count = 0
+    bingo_you_success = False
+    with open(filename, "r") as log_file:
+        for line in log_file:
+            count += 1
+            old_lines.append(line)
+            if "bingo ....you success" in line:
+                bingo_you_success = True
+            if 'ERROR' in line:
+                okay = False
+                errors.append((count, filename, old_lines))
+    error_lines = ""
+    if not okay:
+        for item in errors:
+            error_lines += "file:{}, line {}, 10 subsequent lines were {}".format(item[1], item[0], item[2])
```

Comment:
My (potentially incorrect) understanding is that "old_lines" deque thing will have the 10 lines up to and including the ERROR line. If that's correct, then should change "subsequent" to "preceeding", otherwise I think it's confusing.
```suggestion
            error_lines += "file:{}, line {}, 10 preceeding lines were {}".format(item[1], item[0], item[2])
```
Of course, if I'm wrong, please ignore.